src/ImageSharp/Formats/Tiff/TiffBitsPerSampleExtensions.cs

```diff
@@ -48,6 +50,13 @@ public static TiffBitsPerSample GetBitsPerSample(this ushort[] bitsPerSample)
                         return TiffBitsPerSample.Bit24;
                     }
 
+                    if (bitsPerSample[0] == TiffConstants.BitsPerSampleRgb4Bit[0] &&
+                        bitsPerSample[1] == TiffConstants.BitsPerSampleRgb4Bit[1] &&
+                        bitsPerSample[2] == TiffConstants.BitsPerSampleRgb4Bit[2])
```

Comment:
```suggestion
                    if (bitsPerSample[2] == TiffConstants.BitsPerSampleRgb4Bit[2] &&
                        bitsPerSample[1] == TiffConstants.BitsPerSampleRgb4Bit[1] &&
                        bitsPerSample[0] == TiffConstants.BitsPerSampleRgb4Bit[0])
```
By flipping the order the JIT only needs to emit bound checks for index 2, thus saving some branches.
You can do the same for `Bit24` above.

I believe for .NET 6 JIT got some improvements in that area, but older runtimes will profit from that change.